scanner/decisions/0004-scannerv4-read-vulneralbilities.md

```diff
@@ -0,0 +1,24 @@
+# 0004 - Scanner V4 NVD CVSS enrichment data updating pipeline
+
+- **Author(s):** Yi Li
+- **Created:** [2023-09-11]
+
+## Context
+
+We are in the process of transitioning the StackRox Scanner to utilize ClairCore as its primary scanning engine. Our objective is twofold: firstly, to allow Central to fetch NVD CVSS data, and secondly, to ensure the Scanner V4 seamlessly connecting with Central, triggering data updates and enriching vulnerabilities with CVSS data. Throughout this pipeline, the Claircore enricher will play a pivotal role. It will manage tasks including downloading CVSS data from Google storage, parsing the information, and updating the data accordingly for Scanner V4
+
+## Decision
+
```

Comment:
1. Scanner is starting, no CVSS has been pulled before -> download JSON always happens.
2. Scanner has donwloaded it before, but timestamp in storage is older (not modififer) -> download JSON don't happen.
3. Scanner has downlodaded it before, but timstamp in storage is newer (modified) -> dowload WILL happen.

Reply:
Add how we download the json.gz for each year. Are we downloading the whole repo or send 20 http requests.

Reply:
We are going to have the CVSSUpdater and it will pull CVSS data from google storage and will generate a consolidated enrichment bundle in central as a goRoutine.